src/components/navigation-bar/navigation-bar.test.js

```diff
@@ -0,0 +1,180 @@
+  return <NavigationBar {...props}>{isChildren}</NavigationBar>;
+};
+
+context("Testing NavigationBar component", () => {
+  describe("should render NavigationBar component", () => {
+    it.each(specialCharacters)(
+      "should render NavigationBar component with %s as a children",
+      (childrenValue) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent>{childrenValue}</NavigationBarComponent>
+        );
+
+        navigationBar().should("have.text", childrenValue);
+      }
+    );
+
+    it("should render NavigationBar component with ariaLabel prop", () => {
+      CypressMountWithProviders(
+        <NavigationBarComponent ariaLabel={testData} />
+      );
+
+      navigationBar().should("have.attr", "aria-label", testData);
+    });
+
+    it.each(variants)(
+      "should render NavigationBar component with %s as a navigationType",
+      (navigationType, color) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent navigationType={navigationType} />
+        );
+
+        navigationBar().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      [true, "not.exist"],
+      [false, "be.visible"],
+    ])(
+      "should render NavigationBar component with isLoading prop set to %s",
+      (boolean, assertion) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent isLoading={boolean}>
+            {testData}
+          </NavigationBarComponent>
+        );
+
+        cy.contains(testData).should(assertion);
+      }
+    );
+
+    it.each(["fixed", "sticky"])(
+      "should render NavigationBar component with position prop set to %s with orientation set to top",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="top" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    it.each(["fixed", "sticky"])(
+      "should render NavigationBar component with position prop set to %s with orientation set to bottom",
+      (position) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent position={position} orientation="bottom" />
+        );
+
+        navigationBar().should("have.css", "position", position);
+      }
+    );
+
+    describe.each(["top", "bottom"])("orientation %s", (orientation) => {
+      it.each(offsetVal)(
+        "should render NavigationBar component with offset prop set to %s px",
+        (offset) => {
+          CypressMountWithProviders(
+            <NavigationBarComponent
+              offset={`${offset}px`}
+              orientation={orientation}
+              position="fixed"
+            />
+          );
+          navigationBar().then(($el) => {
+            expect($el).to.have.attr("offset").to.equals(`${offset}px`);
+            expect($el).to.have.css(orientation).to.equals(`${offset}px`);
+          });
```

Comment:
```suggestion
          navigationBar().then(($el) => {
            expect($el).to.have.css(orientation).to.equals(`${offset}px`);
          });
```
I don't think this line is needed since `offset` is a custom React prop here. It seems like @robinzigmond mentions that we may have a separate bug where an `offset` attribute is being added onto the `<nav>` element. This shouldn't affect the visual functionality but it would be good to investigate why this is being added to the DOM element.